### PR TITLE
Seqref updates

### LINKED ISSUE #567: SequenceReference molecule type
We should add molecule type for SequenceReference: Genomic, Transcript, Protein

### LINKED ISSUE #568: SequenceReference literal
We have a use case for the MAVE mapping data where we want to describe a sequence reference _with_ a corresponding literal sequence.

We should add a `sequence` field (analogous to the same concept in `sequenceLocation`) that allows us to explicitly describe the literal sequence.

### PR DESCRIPTION
**Pull request by ahwagner**
Addresses #567 and #568. For discussion.

### CODE REVIEW

File: schema/vrs/vrs-source.yaml [hunk 1]
```diff
@@ -343,6 +343,17 @@ $defs:
         enum:
           - aa
           - na
+      sequence:
+        type:
+          $ref: "#/$defs/SequenceString"
+        description: >-
+          A :ref:`SequenceString` that is a literal representation of the referenced sequence.
+      moleculeType:
```
Comment (ahwagner) [inline schema/vrs/vrs-source.yaml:358]: @cmprocknow concern about "molecule type"; perhaps "molecularContext" would be cleaner.

File: schema/vrs/vrs-source.yaml [hunk 2]
```diff
@@ -343,6 +343,17 @@ $defs:
...
+        type: string
+        enum:
+          - genomic
```
Comment (ahwagner) [inline schema/vrs/vrs-source.yaml:363]: general discussion on genomic expressed concerns, possible alternate is "template"

Comment (ahwagner) [reply schema/vrs/vrs-source.yaml:363]: except "template" is a total non-starter. There are genomic variants that are not necessarily going to be involved as template sequence (e.g. variants in regulatory elements)

Comment (ahwagner) [reply schema/vrs/vrs-source.yaml:363]: Group consensus: we should cover the same concept types that RefSeq uses.

File: schema/vrs/def/SequenceReference.rst [hunk 3]
```diff
@@ -82,6 +82,16 @@ Some SequenceReference attributes are inherited from :ref:`gks-core:Entity`.
       - string
       - 0..1
       - The interpretation of the character codes referred to by the refget accession, where "aa" specifies an amino acid character set, and "na" specifies a nucleic acid character set.
+   *  - sequence
+      - 
+      - {'$ref': '#/$defs/sequenceString'}
+      - 0..1
+      - A :ref:`SequenceString` that is a literal representation of the referenced sequence.
```
Comment (larrybabb) [inline schema/vrs/def/SequenceReference.rst]: This should be lowercase `sequenceString`...

File: schema/vrs/def/SequenceReference.rst [hunk 4]
```diff
@@ -82,6 +82,16 @@ Some SequenceReference attributes are inherited from :ref:`gks-core:Entity`.
...
+   *  - moleculeType
+      - 
+      - string
```
Comment (larrybabb) [inline schema/vrs/def/SequenceReference.rst:92]: MappableConcept? or are you good with an enum here? I defer to your judgement on this. I almost think we should avoid introducing MappableConcept into VRS unless it really needs to happen.

File: schema/vrs/vrs-source.yaml (hunk 1 above)
Comment (larrybabb) [reply schema/vrs/vrs-source.yaml:358]: I think we can sort this out during public review.

File: schema/vrs/vrs-source.yaml (hunk 2 above)
Comment (larrybabb) [reply schema/vrs/vrs-source.yaml:363]: lgtm

File: schema/vrs/def/SequenceReference.rst (hunk 4 above)
Comment (larrybabb) [reply schema/vrs/def/SequenceReference.rst:92]: forget this. enum works for this one.

**Review by larrybabb — CHANGES_REQUESTED** [review]

File: schema/vrs/def/SequenceReference.rst (hunk 3 above)
Comment (ahwagner) [reply schema/vrs/def/SequenceReference.rst]: good catch!

File: schema/vrs/vrs-source.yaml (hunk 1 above)
Comment (ahwagner) [reply schema/vrs/vrs-source.yaml:358]: RefSeq calls this Molecule Type, so seems appropriate to keep.

**Pull request merged**